src/gaussnewton.jl

```diff
@@ -99,12 +113,20 @@ end
 function perform_step!(cache::GaussNewtonCache{true})
     @unpack u, fu1, f, p, alg, J, JᵀJ, Jᵀf, linsolve, du = cache
     jacobian!!(J, cache)
-    __matmul!(JᵀJ, J', J)
-    __matmul!(Jᵀf, J', fu1)
 
-    # u = u - J \ fu
-    linres = dolinsolve(alg.precs, linsolve; A = __maybe_symmetric(JᵀJ), b = _vec(Jᵀf),
-        linu = _vec(du), p, reltol = cache.abstol)
+    if JᵀJ !== nothing
+        __matmul!(JᵀJ, J', J)
+        __matmul!(Jᵀf, J', fu1)
+    end
+
+    # u = u - JᵀJ \ Jᵀfu
+    if cache.JᵀJ === nothing
+        linres = dolinsolve(alg.precs, linsolve; A = J, b = _vec(fu1), linu = _vec(du),
```

Comment:
I get an error stemming from here because the Jacobian is not square

Reply:
what is the `linsolve.alg`? Is `A` or `b` some non-standard array type?

Reply:
Also does specifying `linsolve = QRFactorization()` fix it?

Reply:
`linsolve.alg` is `nothing`. Changing to `QRFactorization()` gives a `not implemented `error. There could be some weird types yes, I'll try purging them...

Reply:
Still doesn't work with only standard arrays

Reply:
Can you post the code for that?